### PR TITLE
[WIP] Min Max DataMap Example

### PR DESCRIPTION
**Pull request by sounakr**
Min Max DataMap Example

### CODE REVIEW

**Comment by jackylk** [comment]
Great, thanks for working on this.
My first comment is that can you move this example to example module, since "letting developer extend datamap in their own library" is one of the goal of datamap framework. We should verify the datamap framework toward this goal.

File: core/src/main/java/org/apache/carbondata/core/cache/CacheType.java [hunk 1]
```diff
@@ -61,6 +61,9 @@
   public static final CacheType<TableSegmentUniqueIdentifier, SegmentTaskIndexWrapper>
       DRIVER_BLOCKLET_DATAMAP = new CacheType("driver_blocklet_datamap");
 
+  public static final CacheType<TableSegmentUniqueIdentifier, SegmentTaskIndexWrapper>
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/cache/CacheType.java]: Don't add cache type, no need to handle cache in example

File: core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/DataMapMinMax.java [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+import org.apache.carbondata.core.scan.filter.FilterUtil;
+import org.apache.carbondata.core.scan.filter.executer.FilterExecuter;
+import org.apache.carbondata.core.scan.filter.resolver.FilterResolverIntf;
+import org.apache.carbondata.core.util.CarbonUtil;
+import org.apache.carbondata.core.util.DataFileFooterConverter;
+
+/**
+ * Datamap implementation for blocklet.
+ */
+public class DataMapMinMax implements DataMap, Cacheable {
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/DataMapMinMax.java]: No need to use Unsafe in example, don't confuse users with unsafe in example. just make it as simple as possibe

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/DataMapMinMaxWriter.scala [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+import org.apache.carbondata.core.datamap.{DataMapStoreManager, TableDataMap}
+import org.apache.carbondata.core.datamap.dev.DataMapWriter
+import org.apache.carbondata.core.datastore.page.ColumnPage
+import org.apache.carbondata.core.metadata.AbsoluteTableIdentifier
+import org.apache.carbondata.core.metadata.datatype.DataType
+import org.apache.carbondata.core.util.CarbonProperties
+
+
+class DataMapMinMaxWriter extends QueryTest with BeforeAndAfterAll {
```
Comment (ravipesala) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/DataMapMinMaxWriter.scala]: Append testcase to the file name

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/MinMaxDataMap.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+  @Override public List<Blocklet> prune(FilterResolverIntf filterExp) {
+    return null;
+  }
+
+  @Override public void clear() {
+
+  }
+
+  public void updateMinMaxIndex(String filePath) throws IOException, MemoryException {
```
Comment (ravipesala) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/MinMaxDataMap.java]: it shouldn't be read from index file. Use the Datamap writer to write example index file and read here

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/9/

**Pull request closed without merging**